fxola/Banka
# #165163326 Setup App for deployment

#### What does this PR do?
- Setup App for deployment, Write additional tests, Refactor validation


#### Description of Task to be completed?
- install dependencies for code coverage generation
- add coverage generation scripts to package.json file
- write extra test cases for uncovered lines
- refactor frequently used validation methods to helper class
- setup travis.yml file
- add build scripts to package.json file
- add Procfile for heroku deployment
- delivers[#165163326]

#### How should this be manually tested?
- Clone the repository using ```git clone https://github.com/fxola/Banka.git .``` in your terminal
- run ```npm install``` on your terminal
- run ```npm run start``` to start up the server
- run ```npm test``` to see test results
- run ```npm run coverage``` to see coverage report
#### Any background context you want to provide?
- Create and environment variable with the format ```SECRET = yoursecret``` in the root of the server directory


#### What are the relevant pivotal tracker stories?
- PT Story link - [#165163326] (https://www.pivotaltracker.com/n/projects/2321791/stories/165163326)

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A
